lib/Swagger/src/Discussion/Api/SitesApi.php

```diff
@@ -334,4 +304,215 @@ public function getSiteWithHttpInfo($site_id)
+        }
+        // path params
+        if ($site_id !== null) {
+            $resourcePath = str_replace(
+                "{" . "siteId" . "}",
+                $this->apiClient->getSerializer()->toPathValue($site_id),
+                $resourcePath
+            );
+        }
+        // default format to json
+        $resourcePath = str_replace("{format}", "json", $resourcePath);
+
+        
+        // for model (json/xml)
+        if (isset($_tempBody)) {
+            $httpBody = $_tempBody; // $_tempBody is the method argument, if present
+        } elseif (count($formParams) > 0) {
+            $httpBody = $formParams; // for HTTP post (form)
+        }
+        // this endpoint requires API key authentication
+        $apiKey = $this->apiClient->getApiKeyWithPrefix('X-Wikia-AccessToken');
+        if (strlen($apiKey) !== 0) {
+            $headerParams['X-Wikia-AccessToken'] = $apiKey;
+        }
+        // this endpoint requires API key authentication
+        $apiKey = $this->apiClient->getApiKeyWithPrefix('X-Wikia-UserId');
+        if (strlen($apiKey) !== 0) {
+            $headerParams['X-Wikia-UserId'] = $apiKey;
+        }
+        // make the API Call
+        try {
+            list($response, $statusCode, $httpHeader) = $this->apiClient->callApi(
+                $resourcePath,
+                'DELETE',
+                $queryParams,
+                $httpBody,
+                $headerParams,
+                '\Swagger\Client\Discussion\Models\SiteHalResponse',
+                '/sites/{siteId}/hard-delete'
+            );
+
+            return array($this->apiClient->getSerializer()->deserialize($response, '\Swagger\Client\Discussion\Models\SiteHalResponse', $httpHeader), $statusCode, $httpHeader);
+        } catch (ApiException $e) {
+            switch ($e->getCode()) {
+                case 204:
+                    $data = $this->apiClient->getSerializer()->deserialize($e->getResponseBody(), '\Swagger\Client\Discussion\Models\SiteHalResponse', $e->getResponseHeaders());
+                    $e->setResponseObject($data);
+                    break;
+                case 403:
+                    $data = $this->apiClient->getSerializer()->deserialize($e->getResponseBody(), '\Swagger\Client\Discussion\Models\HalProblem', $e->getResponseHeaders());
+                    $e->setResponseObject($data);
+                    break;
+            }
+
+            throw $e;
+        }
+    }
+
+    /**
+     * Operation softDeleteSite
+     *
+     * Soft delete
+     *
+     * @param int $site_id The id of the site (required)
+     * @param string $x_wikia_schwartz  (optional)
+     * @return \Swagger\Client\Discussion\Models\SiteHalResponse
+     * @throws \Swagger\Client\ApiException on non-2xx response
+     */
+    public function softDeleteSite($site_id, $x_wikia_schwartz = null)
+    {
+        list($response) = $this->softDeleteSiteWithHttpInfo($site_id, $x_wikia_schwartz);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$response". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)